File: hackerrank/easy/0055-lisa-workbook/js/solution.mjs
// solution.mjs

export function workbook(n, k, arr) {
  let page = 0;
  let specialCount = 0;

  for (let chapter = 0; chapter < n; chapter++) {
    const problems = arr[chapter];
    const startPage = page + 1;
    const totalPages = Math.floor((problems + k - 1) / k);
    page += totalPages;

    for (let p = 1; p <= problems; p++) {
      const pageNumber = startPage + Math.floor((p - 1) / k);
      if (p === pageNumber) {
        specialCount++;
      }
    }
  }

  return specialCount;
}

// --- demo run ---
if (import.meta.url === `file://${process.argv[1]}`) {
  const nSample = 5;
  const kSample = 3;
  const arrSample = [4, 2, 6, 1, 10]; // Expected: 4
  console.log(workbook(nSample, kSample, arrSample));
}

File: hackerrank/easy/0055-lisa-workbook/python/solution.py
## solution.py

from typing import List


def workbook(n: int, k: int, arr: List[int]) -> int:
    page = 0
    special_count = 0

    for problems in arr:
        start_page = page + 1
        total_pages = (problems + k - 1) // k
        page += total_pages

        for p in range(1, problems + 1):
            page_number = start_page + (p - 1) // k
            if p == page_number:
                special_count += 1

    return special_count


# --- demo run ---
if __name__ == "__main__":
    n_sample = 5
    k_sample = 3
    arr_sample = [4, 2, 6, 1, 10]  # Expected: 4
    print(workbook(n_sample, k_sample, arr_sample))
